providers/async_gemini.py

```
@@ -0,0 +1,534 @@
+        # Add images if provided and supported
+        if images and self._supports_vision(resolved_name):
+            for image_path in images:
+                try:
+                    image_content = await self._process_image_async(image_path)
+                    if image_content:
+                        content_parts.append(image_content)
+                except Exception as e:
+                    logger.warning(f"Failed to process image {image_path}: {e}")
+                    continue
+        elif images and not self._supports_vision(resolved_name):
+            logger.warning(f"Model {resolved_name} does not support images, ignoring {len(images)} image(s)")
```

Pylint (3.3.7)

[convention] Line too long (109/100)

(C0301)